Dockerfile.cuda-all

```
@@ -1,49 +1,65 @@
 # syntax=docker/dockerfile:1
 
-FROM nvidia/cuda:12.4.1-cudnn-devel-ubuntu22.04 AS builder
+FROM docker.io/nvidia/cuda:12.9.1-cudnn-devel-ubuntu22.04 AS builder
 
 ARG DEBIAN_FRONTEND=noninteractive
 RUN <<HEREDOC
     apt-get update
     apt-get install -y --no-install-recommends \
         curl \
         libssl-dev \
-        pkg-config
+        pkg-config && \
 
     rm -rf /var/lib/apt/lists/*
 HEREDOC
 
-RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y && test -f /root/.cargo/bin/rustup
```

The [associated commit](https://github.com/EricLBuehler/mistral.rs/pull/1468/files/4d45bfd83cd8e5838ce9d7ff916c45deb3c07fe5..444ddc01f0625e6be5d080bedf5cc993f3eab9f0) for the `test` addition here is about exit codes? Did you encounter a failure?

If it was due to `curl` itself failing which I assume was more likely, then it's more about how bash works with the pipe operator by default.

```suggestion
SHELL ["/bin/bash", "-e", "-o", "pipefail", "-c"]
RUN curl -fsSL https://sh.rustup.rs | bash -s -- -y
```

Typically the `SHELL` instruction would be set earlier than this suggestion places it, and it will instead default `RUN` instructions to use that command instead of the default `/bin/sh -c` command.

However since you have brought up podman in the past, for those that would build a `Dockerfile` with this instruction it is not part of OCI standard spec IIRC, so Podman needs an extra opt-in flag for it's build command (_it'll otherwise fail and point that problem as the failure cause_). Minor concern though given the improvement to the build process.